### ISSUE TITLE
Proposal: ColorSwatches in ColorPicker

### ISSUE THREAD

**Issue by chingucoding** [microsoft/microsoft-ui-xaml#1659]
![](https://user-images.githubusercontent.com/7389110/69379335-34513e00-0ca8-11ea-8969-6e3755371314.png)
Credit: @mdtauk
# Proposal: ColorSwatches in ColorPicker

## Summary

The win32 ColorPicker has swatches, the WinUI colorpicker should too!

## Rationale

* Close gap between win32 and WinUI
* Ability for developers to let users input accent colors easily
* Enables history of colors chosen by user

## Scope

| Capability | Priority |
| :---------- | :------- |
| Close gap between win32 ColorPicker and WinUI ColorPicker | Must |
| List of windows accent colors | Must |
| Possibility to disable swatches | Must |
| Let developer disable list of accent colors | Must |
| List of app colors (e.g. list of brand colors) | Must |
| Disable list of app colors | Must |
| Change text of "App colors" label | Must |
| History of colors chosen by user | Should |
| Let developer disable list of color history | Should |
| Host custom XAML content | Could |

## Important Notes

Idea: @mdtauk [(this comment)](https://github.com/microsoft/microsoft-ui-xaml-specs/pull/53#issuecomment-557086590)
## Open Questions

If we disable everything in the swatches panel, what will we display?

**Comment by mdtauk** [comment]
Thank you for writing this up as a proper proposal @chingucoding I could imagine there may be scope for further "sections" that could be added to the ColorPicker, if some compelling ideas were to be proposed.  Failing that, there could be custom XAML content added.

**Comment by shaheedmalik** [comment]
Yes. It looks great.

**Comment by adrientetar** [comment]
Why not put R,G,B on the same line and the legends below?

**Comment by chingucoding** [comment]
> Why not put R,G,B on the same line and the legends below?

There were also some thoughts put into this idea in #1637. The main problem I saw is how to deal with lengthy translations.

**Comment by robloo** [comment]
>>Why not put R,G,B on the same line and the legends below?

>There were also some thoughts put into this idea in #1637. The main problem I saw is how to deal with lengthy translations.

Yes, when I added the suggestion I also mentioned translation could be an issue. However, there are some methods of fixing this:
 1. The text could have trimming
 2. The control could resize itself
 3. Just use color component abbreviations like Photoshop(r) has: just 'R', 'G', 'B' 

I like #3 the best and would argue if the user is editing the components directly they already know what they are doing. I'll add this comment in the other discussion too.

As far as color swatches, I also added this in one of my apps. Might I suggest a layout like what is below? It also has the 'standard' I'll say 16 colors which should be an option some place.

![image](https://user-images.githubusercontent.com/17993847/69482567-16d8bd00-0deb-11ea-9a4d-a1e07fea3816.png)

**Comment by chingucoding** [comment]
> As far as color swatches, I also added this in one of my apps. Might I suggest a layout like what is below? It also has the 'standard' I'll say 16 colors which should be an option some place. Another useful add is the 'Clear' button which will set the alpha channel to zero for full transparency. Transparency should be a selectable option somehow.

This would also make sense. Maybe color swatches could have a DisplayMode with values "Hidden", "Inline" and "Tab".

**Comment by mdtauk** [comment]
Posting this by way of an example of another Fluent based UI kit from Microsoft doing Color Picking and Swatches...

![Fabric Web - Color Pickers](https://user-images.githubusercontent.com/7389110/69777501-7b539d80-1198-11ea-8890-4f5099f65c78.png)

**Comment by mdtauk** [comment]
Originally posted in the ColorPicker Orientation Issue, her is a horizontal and vertical mockup for how Swatches could be implemented.

![image](https://user-images.githubusercontent.com/7389110/77865094-770dbe80-7224-11ea-910a-ae8323a0e3f6.png)
